### PR TITLE
Make transformation docstring more precise

### PR DESCRIPTION
**Pull request by bkamins**

### CODE REVIEW

File: src/abstractdataframe/selection.jl [hunk 1]
```diff
@@ -57,7 +57,9 @@ const TRANSFORMATION_COMMON_RULES =
        it must be single name (as a `Symbol` or a string).
     6. vectors or matrices containing transformations specified by the `Pair` syntax
        described in points 2 to 5
-    8. a function which will be called with a `SubDataFrame` corresponding to each group;
+    7. a function which will be called with a `SubDataFrame` corresponding to each group
+       when `GroupedDataFrame` is processed or the passed source data frame if
+       `AbstractDataFrame` is processed;
```
Comment (nalimilan) [inline src/abstractdataframe/selection.jl:62]:
```suggestion
       if a `GroupedDataFrame` is processed, or with the data frame itself if
       an `AbstractDataFrame` is processed;
```

File: docs/src/man/split_apply_combine.md [hunk 2]
```diff
@@ -63,7 +63,9 @@ each subset of the `DataFrame`. This specification can be of the following forms
    it must be single name (as a `Symbol` or a string).
 6. vectors or matrices containing transformations specified by the `Pair` syntax
    described in points 2 to 5
-8. a function which will be called with a `SubDataFrame` corresponding to each group;
+7. a function which will be called with a `SubDataFrame` corresponding to each group
+   when `GroupedDataFrame` is processed or the passed source data frame if
+   `AbstractDataFrame` is processed;
```
Comment (nalimilan) [inline docs/src/man/split_apply_combine.md:68]: This section only covers split-apply-combine, so the input is supposed to be a `GroupedDataFrame` AFAICT. Though I guess we'd better keep these in sync for consistency?

Comment (bkamins) [reply docs/src/man/split_apply_combine.md:68]: Yes - this is what I think. Also in general at some point this section should be rewritten at some point not to make such a strong distinction between split-apply-combine and other transformations as currently they are the same conceptually.

Comment (bkamins) [inline docs/src/man/split_apply_combine.md:68]:
```suggestion
   if a `GroupedDataFrame` is processed, or with the data frame itself if
   an `AbstractDataFrame` is processed;
```

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**